iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java

```diff
@@ -0,0 +1,139 @@
+    DELETE_FILE_READ_META_COLS.put(MetadataColumns.SPEC_ID, 0);
+    DELETE_FILE_READ_META_COLS.put(PARTITION_STRUCT_META_COL, 1);
+    DELETE_FILE_READ_META_COLS.put(MetadataColumns.FILE_PATH, 2);
+    DELETE_FILE_READ_META_COLS.put(MetadataColumns.ROW_POSITION, 3);
+  }
+
+  private static final Types.NestedField PARTITION_HASH_META_COL = Types.NestedField.required(
+      MetadataColumns.PARTITION_COLUMN_ID, MetadataColumns.PARTITION_COLUMN_NAME, Types.LongType.get());
+  private static final Map<Types.NestedField, Integer> DELETE_SERDE_META_COLS = Maps.newLinkedHashMap();
+
+  static {
+    DELETE_SERDE_META_COLS.put(MetadataColumns.SPEC_ID, 0);
+    DELETE_SERDE_META_COLS.put(PARTITION_HASH_META_COL, 1);
+    DELETE_SERDE_META_COLS.put(MetadataColumns.FILE_PATH, 2);
+    DELETE_SERDE_META_COLS.put(MetadataColumns.ROW_POSITION, 3);
+  }
+
+  /**
+   * @param dataCols The columns of the original file read schema
+   * @param table The table object - it is used for populating the partition struct meta column
+   * @return The schema for reading files, extended with metadata columns needed for deletes
+   */
+  public static Schema createFileReadSchemaForDelete(List<Types.NestedField> dataCols, Table table) {
+    List<Types.NestedField> cols = Lists.newArrayListWithCapacity(dataCols.size() + DELETE_FILE_READ_META_COLS.size());
+    DELETE_FILE_READ_META_COLS.forEach((metaCol, index) -> {
+      if (metaCol == PARTITION_STRUCT_META_COL) {
+        cols.add(MetadataColumns.metadataColumn(table, MetadataColumns.PARTITION_COLUMN_NAME));
+      } else {
+        cols.add(metaCol);
+      }
+    });
+    cols.addAll(dataCols);
+    return new Schema(cols);
+  }
+
+  /**
+   * @param dataCols The columns of the serde projection schema
+   * @return The schema for SerDe operations, extended with metadata columns needed for deletes
+   */
+  public static Schema createSerdeSchemaForDelete(List<Types.NestedField> dataCols) {
+    List<Types.NestedField> cols = Lists.newArrayListWithCapacity(dataCols.size() + DELETE_SERDE_META_COLS.size());
+    DELETE_SERDE_META_COLS.forEach((metaCol, index) -> cols.add(metaCol));
+    cols.addAll(dataCols);
+    return new Schema(cols);
+  }
+
+  /**
+   * @param rec The record read by the file scan task, which contains both the metadata fields and the row data fields
+   * @param rowData The record object to populate with the rowData fields only
+   * @return The position delete object
+   */
+  public static PositionDelete<Record> getPositionDelete(Record rec, GenericRecord rowData) {
+    PositionDelete<Record> positionDelete = PositionDelete.create();
+    String filePath = rec.get(DELETE_SERDE_META_COLS.get(MetadataColumns.FILE_PATH), String.class);
+    long filePosition = rec.get(DELETE_SERDE_META_COLS.get(MetadataColumns.ROW_POSITION), Long.class);
+
+    int dataOffset = DELETE_SERDE_META_COLS.size(); // position in the rec where the actual row data begins
+    for (int i = dataOffset; i < rec.size(); ++i) {
```

Comment:
Do we want to have a config for writing deletes without the original data? It would result in faster writes (less data to read, and less data to write), but we might lose some read performance because of missing statistics.

ACID tables write only the metadata.

Maybe a table level, plus a global config. Maybe in a follow-up PR

Reply:
This is a good idea. Let's make this configurable. I will address this in a separate PR.